akka-cluster/src/test/scala/akka/cluster/ClusterDomainEventSpec.scala

```diff
@@ -37,6 +36,15 @@ class ClusterDomainEventSpec extends AnyWordSpec with Matchers {
   val eDown = TestMember(Address("akka", "sys", "e", 2552), Down, eRoles)
   val selfDummyAddress = UniqueAddress(Address("akka", "sys", "selfDummy", 2552), 17L)
 
+  private val originalClusterAssert = sys.props.get("akka.cluster.assert").getOrElse("false")
+  override protected def beforeAll(): Unit = {
+    System.setProperty("akka.cluster.assert", "on")
+  }
+
+  override protected def afterAll(): Unit = {
+    System.setProperty("akka.cluster.assert", originalClusterAssert)
+  }
+
```

Comment:
What is this? Is that something we pass in Jenkins? 

If so, maybe some notes to clarify why we are doing this will be helpful since this is not part of the code base.

Reply:
@johanandren, I found a recent comment from you in Slack and found some other references in the code base. 

TIL as well

Reply:
On the other hand, why are you enabling this? 

None of the tests you added seems to be using it. Or am I missing something?

Reply:
I noticed that it leaks between tests, so in CI or running all tests it is always enabled, but when running this single test it is not, so I enabled it to get consistent testing. The flag toggles an invariant assertion inside of Gossip that I saw fail only on CI which is how I noticed it.

Reply:
(All tests here create Gossip instances)